### ISSUE TITLE
error message related to censusdate

### ISSUE THREAD

**Issue by patdumandan** [weecology/portalcasting#187]
@juniperlsimonis , This error message pops up: *Error: Can't combine `..1$censusdate` <character> and `..2$censusdate` <date>.*  when the following functions are run:  
* setup_dir(main=main)  
* prep_moons(main = main, lead_time = 12, cast_date = Sys.Date(), control_files = files_control(), quiet = TRUE, verbose = FALSE, arg_checks = TRUE) 
* plot_cast_ts() 

when, rlang::last_error() is ran, this is what we see: dplyr::bind_rows(moons, .), and then a lot of stop_incompatible() messages

**Comment by juniperlsimonis** [comment]
huh, weird! works fine for me still... can you run `sessionInfo()` please?

**Comment by patdumandan** [comment]
![image](https://user-images.githubusercontent.com/46734193/96777094-db0df500-13b7-11eb-8842-bf2b4e0dec7f.png)

This is what I get, @juniperlsimonis

**Comment by juniperlsimonis** [comment]
@skmorgane and @ethanwhite if you all get the same error, can you run `sessionInfo()` as well please? thanks!

**Comment by ethanwhite** [comment]
Yep. Here's the error I get at the tail end of running `setup_production()`:

```r
> setup_production()
...
cast_id_999_model_fits.json
casts_metadata.csv
   **5299 casts files moved**
 -Writing model scripts
 -Writing model scripts
  -Adding AutoArima
 -Writing model scripts
  -Adding ESSS
 -Writing model scripts
  -Adding NaiveArima
 -Writing model scripts
  -Adding nbGARCH
 -Writing model scripts
  -Adding nbsGARCH
 -Writing model scripts
  -Adding pevGARCH
 -Writing model scripts
  -Adding simplexEDM
 -Writing model scripts
  -Adding GPEDM
 -Writing model scripts
  -Adding jags_RW
 -Adding data files to data subdirectory
  -lunar data file
Loading in data version 2.48.0
Error: Can't combine `..1$censusdate` <character> and `..2$censusdate` <date>.
Run `rlang::last_error()` to see where the error occurred.
```

And the associated output from `sessionInfo`

```r
> sessionInfo()
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Pop!_OS 20.04 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] portalcasting_0.18.3

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.5        ltsa_1.4.6        urca_1.3-0        pillar_1.4.6     
 [5] compiler_3.6.3    viridis_0.5.1     runjags_2.0.4-6   tseries_0.10-47  
 [9] xts_0.12.1        tools_3.6.3       viridisLite_0.3.0 lubridate_1.7.9  
[13] jsonlite_1.7.1    lifecycle_0.2.0   tibble_3.0.4      gtable_0.3.0     
[17] nlme_3.1-144      lattice_0.20-40   pkgconfig_2.0.3   rlang_0.4.8      
[21] yaml_2.2.1        curl_4.3          parallel_3.6.3    gridExtra_2.3    
[25] coda_0.19-4       lunar_0.1-04      dplyr_1.0.2       httr_1.4.2       
[29] hms_0.5.3         generics_0.0.2    vctrs_0.3.4       lmtest_0.9-38    
[33] grid_3.6.3        nnet_7.3-13       forecast_8.13     tidyselect_1.1.0 
[37] glue_1.4.2        R6_2.4.1          readr_1.4.0       portalr_0.3.5    
[41] ggplot2_3.3.2     purrr_0.3.4       TTR_0.24.2        magrittr_1.5     
[45] scales_1.1.1      ellipsis_0.3.1    tscount_1.4.3     quantmod_0.4.17  
[49] timeDate_3043.102 colorspace_1.4-1  fracdiff_1.5-1    quadprog_1.5-8   
[53] munsell_0.5.0     crayon_1.3.4      zoo_1.8-8      
```

And the `rlang::last_error()` trace:

```r
> rlang::last_error
function () 
{
    if (is_null(last_error_env$cnd)) {
        abort("Can't show last error because no error was recorded yet")
    }
    cnd <- last_error_env$cnd
    cnd$rlang$internal$from_last_error <- TRUE
    cnd
}
<bytecode: 0x55bffad91c38>
<environment: namespace:rlang>
> rlang::last_error()
<error/vctrs_error_incompatible_type>
Can't combine `..1$censusdate` <character> and `..2$censusdate` <date>.
Backtrace:
  1. portalcasting::setup_production()
  2. portalcasting::setup_dir(...)
  3. portalcasting::fill_dir(...)
  4. portalcasting::fill_data(...)
  5. portalcasting::prep_moons(...)
  6. portalcasting::add_future_moons(...)
  7. portalr::get_future_moons(moons, lead_time)
  7. portalcasting::add_extra_future_moons(., cast_date)
  7. dplyr::mutate(., newmoondate = as.character(newmoondate))
 15. dplyr::bind_rows(moons, .)
 16. vctrs::vec_rbind(!!!dots, .names_to = .id)
 18. vctrs::vec_default_ptype2(...)
 19. vctrs::stop_incompatible_type(...)
 20. vctrs:::stop_incompatible(...)
 21. vctrs:::stop_vctrs(...)
Run `rlang::last_trace()` to see the full context.
```

**Comment by gmyenni** [comment]
Closed by #189
